### ISSUE TITLE
Typescript error when using CommonJS module outputs

### ISSUE THREAD

**Issue by esuh-descript** [fullcalendar/temporal-polyfill#35]
I am running across this bug when trying to import this package into a Typescript project configured to emit CommonJS modules:

```shell-session
$ pnpm run typecheck
> temporal-polyfill-bug@1.0.0 typecheck /src/temporal-polyfill-bug
> tsc --build tsconfig.json

index.ts:1:26 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("temporal-polyfill")' call instead.
  To convert this file to an ECMAScript module, change its file extension to '.mts', or add the field `"type": "module"` to '/src/temporal-polyfill-bug/package.json'.

1 import { Temporal } from 'temporal-polyfill';
                           ~~~~~~~~~~~~~~~~~~~


Found 1 error.

 ELIFECYCLE  Command failed with exit code 1.
```

I've reproduced the above bug using a [minimal example](https://github.com/descriptinc/temporal-polyfill-bug) by cloning, running `pnpm i`, and then `pnpm run typecheck`.

I believe a fix is relatively straightforward based on microsoft/TypeScript#53045.

**Comment by arshaw** [comment]
Thanks @esuh-descript , I'll make a .cts file for the next version.

**Comment by arshaw** [comment]
This has been fixed in [v0.2.4](https://github.com/fullcalendar/temporal-polyfill/releases/tag/v0.2.4)

@esuh-descript, I embedded your test case [into the codebase's CI](https://github.com/fullcalendar/temporal-polyfill/tree/main/examples/temporal-polyfill-cjs) to ensure things don't break in the future. Thanks for that!
